### DISCUSSION TITLE
Crop a group, an object...

### DISCUSSION THREAD

**Discussion by tobiasBora** [ManimCommunity/manim#2815] [Ideas]
## Description of proposed feature

Sometimes, it may be useful to crop an object, because the object could be theoretically infinite and I may just want to display a part of it, with other animations or text around it. It would be cool to be able to apply it to a group for instance.

## How can the new feature be used?

For instance, I'd like to try to draw the following thing:

![image](https://user-images.githubusercontent.com/2164118/159113347-eb175b8c-f354-4d47-b8ab-823c3f9e2120.png)

If crop is allowed with an arbitrary path, then we could also crop rectangle images to circle, and much fun would be possible.

## Possible workarounds

The only workaround I can think of would be to draw some squares around the image with the color of the background, but it is not perfect: first, it's tedious to write properly, especially when multiple groups must be cropped. Secondly, it will fail if we want to make an object pass behind the crop part, or if two cropped objects are too close to each others. I also tried Intersection, but without luck on groups.

**Comment by kilacoda-old** [comment]
One way I think you can do this by creating a mask type thing with the difference of two rectangles, one being the dimensions of the frame and the other being of the area you want cropped, and setting the fill color to the background color with full fill opacity. Though for multiple objects and related masks you'd have to fiddle around with the z_index and stuff I'd assume.

```py
from manim import *

class Masking(Scene):
    def construct(self):
        numplane = NumberPlane(fill_opacity=0.5)

        self.add(numplane)

        mask = Difference(
            Rectangle(height=config.frame_height, width=config.frame_width), #whole frame
            Rectangle(height=3,width=5),                                     #cropped
            stroke_opacity=0,
            fill_color=BLACK,
            fill_opacity=1
        )

        self.add(mask)

```
![image](https://user-images.githubusercontent.com/65204531/159337842-3e782d40-f271-4ba2-bb4c-980afcd3535d.png)

**Comment by tobiasBora** [comment]
Thanks for this trick. It is what I was referring to as a workaround (thanks for writing this code, it's a quite clean way to do this workaround) as it gets trickier to use when we use multiples cropped objects (the mask of the second object will hide the first object, so we need to remove from it the mask of the first object). And also, it would not be able to handle movements going behind the object.

**Comment by tobiasBora** [comment]
Any reasons this issue has been moved to discussion? It should be considered as a feature request as appropriate crop is not available.

**Reply by behackl** [reply]
All suggestions and feature requests have been moved to discussions to in an effort to clean up and get a proper overview over our backlog. Once there is a concrete proposal for an implementation, an issue can be created.

**Reply by tobiasBora** [reply]
Ok thanks!
